src/python/grpcio/grpc/_cython/_cygrpc/credentials.pyx.pxi

```diff
@@ -380,3 +380,22 @@ def server_credentials_alts():
   # Options can be destroyed as deep copy was performed.
   grpc_alts_credentials_options_destroy(c_options)
   return credentials
+
+
+cdef class ComputeEngineChannelCredentials(ChannelCredentials):
+  cdef grpc_channel_credentials* _c_creds
+  cdef grpc_call_credentials* _call_creds
+
+  def __cinit__(self, CallCredentials call_creds):
+    self._c_creds = NULL
+    self._call_creds = call_creds.c()
+    if self._call_creds == NULL:
```

Comment:
Sorry for digging up such an old PR, but I am curious about this NULL check. grpc_google_default_credentials_create specifically allows an nullptr value for the _call_creds argument (creating a default, according to the docstring and the source code).

There must be a reason we chose to not surfacing this feature via the Python API -- but I could not find out from the review discussion of this PR. Any insights? @gnossen

Reply:
It's been quite a while, so my memory is not fresh here, but IIRC, the reason is that we expect an external Python library to implement the application default credentials flow. The reason a default was added to the core API `grpc_google_default_credentials_create` was to make it more generally useful for languages that do not have a strong expectation that this must be done externally.

Reply:
Thanks for clarification. If I understood the [Google OAuth example](https://grpc.io/docs/guides/auth/#using-google-token-based-authentication) correctly, the C++ API relies on support a NULL value as a signal to create a 'default GCE style' credential.

To achieve the same functionality with grpc python, I will need to find a way to do the equivalent of [make_default_call_creds](https://github.com/grpc/grpc/blob/f04eee5cae7e8c372b54f3e64822b2e2eab89762/src/core/lib/security/credentials/google_default/google_default_credentials.cc#L347).

Reply:
I figured this out. We shall use google.auth.default with google.auth.transport.grpc.AuthMetadataPlugin. There is an example that does this in one of the aio unit tests.